[ci] make ci-coqtail fails when not run from an installed Coq

The script "dev/ci/ci-coqtail.sh" hangs here (I waited several minutes), whereas it runs in a few seconds on the pre-13617 master.  Do I need something else to run it now?

```
=============================================== test session starts ===============================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /home/proj/coq/_build_ci/coqtail
collected 13 items

tests/coq/test_coqtop.py
```

Now the tests run, but they all fail:

```
$ cd /home/proj
$ dir
PythonTest  ceps  coq  coq2  coqdune  empty-dir  gtk
$ ls coqdune/
coq
$ cd coqdune
$ rm -rf *
$ git clone --depth 1 https://github.com/coq/coq.git
Cloning into 'coq'...
remote: Enumerating objects: 4967, done.
remote: Counting objects: 100% (4967/4967), done.
remote: Compressing objects: 100% (4701/4701), done.
remote: Total 4967 (delta 448), reused 2123 (delta 119), pack-reused 0
Receiving objects: 100% (4967/4967), 7.54 MiB | 2.18 MiB/s, done.
Resolving deltas: 100% (448/448), done.
jfehrle@LAPTOP-LB5T3R83:/home/proj/coqdune$ ./configure -profile devel
bash: ./configure: No such file or directory
jfehrle@LAPTOP-LB5T3R83:/home/proj/coqdune$ cd coq
jfehrle@LAPTOP-LB5T3R83:/home/proj/coqdune/coq$ ./configure -profile devel
Creating pre-commit hook in .git/hooks/pre-commit
You have OCaml 4.07.1. Good!
You have OCamlfind 1.8.1. Good!
You have native-code compilation. Good!
You have the Zarith library 1.12 installed. Good!
LablGtk3 and LablGtkSourceView3 found (3.1.1), with native threads:
=> native CoqIde will be built.

  Architecture                : Linux
  Sys.os_type                 : Unix
  Coq VM bytecode link flags  : -dllib -lcoqrun -dllpath "/home/proj/coqdune/coq/kernel/byterun"
  Other bytecode link flags   :
  OCaml version               : 4.07.1
  OCaml binaries in           : /home/jfehrle/.opam/4.07.1/bin/
  OCaml library in            : /home/jfehrle/.opam/4.07.1/lib/ocaml
  OCaml flambda flags         :
  Native dynamic link support : true
  Lablgtk3 library in         : /home/jfehrle/.opam/4.07.1/lib/lablgtk3-sourceview3
  CoqIde                      : opt
  Documentation               : None
  Web browser                 : firefox -remote "OpenURL(%s,new-tab)" || firefox %s &
  Coq web site                : http://coq.inria.fr/
  Bytecode VM enabled         : true
  Native Compiler enabled     : ondemand

  Local build, no installation...

If anything is wrong above, please restart './configure'.

*Warning* To compile the system for a new architecture
          don't forget to do a 'make clean' before './configure'.
$ make -f Makefile.dune world
dune build  coq-core.install coq-stdlib.install coqide-server.install coqide.install
      coqdep user-contrib/Ltac2/Env.v.d
*** Warning: in file Env.v,
    required library Init matches several files in path
    (found Init.v in ../../theories/Classes and .; used the latter)
      coqdep user-contrib/Ltac2/Ind.v.d
*** Warning: in file Ind.v,
    required library Init matches several files in path
    (found Init.v in ../../theories/Classes and .; used the latter)
$ bash dev/ci/ci-coqtail.sh
++ : 1
++ export NJOBS
++ '[' -n '' ']'
++ '[' -d /home/proj/coqdune/coq/_build_vo/ ']'
++ '[' -d /home/proj/coqdune/coq/_build/install/default/ ']'
++ export OCAMLPATH=/home/proj/coqdune/coq/_build/install/default/lib/:/home/proj/coqdune/coq/_install_ci/lib:
++ OCAMLPATH=/home/proj/coqdune/coq/_build/install/default/lib/:/home/proj/coqdune/coq/_install_ci/lib:
++ export COQBIN=/home/proj/coqdune/coq/_build/install/default/bin
++ COQBIN=/home/proj/coqdune/coq/_build/install/default/bin
++ export COQLIB=/home/proj/coqdune/coq/_build/install/default/lib/coq
++ COQLIB=/home/proj/coqdune/coq/_build/install/default/lib/coq
+++ git rev-parse --abbrev-ref HEAD
++ CI_BRANCH=master
++ export CI_BRANCH
++ export 'PATH=/home/proj/coqdune/coq/_build/install/default/bin:/home/proj/coqdune/coq/_install_ci/bin:/home/jfehrle/.opam/4.07.1/bin:/home/jfehrle/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/mnt/c/windows/system32:/mnt/c/windows:/mnt/c/windows/System32/Wbem:/mnt/c/windows/System32/WindowsPowerShell/v1.0/:/mnt/c/windows/System32/OpenSSH/:/mnt/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NGX:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NvDLISR:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/Program Files/Docker/Docker/resources/bin:/mnt/c/ProgramData/DockerDesktop/version-bin:/mnt/c/Gradle/gradle-6.4.1/bin":/mnt/c/Users/jfehr/AppData/Local/Microsoft/WindowsApps:/snap/bin'
++ PATH='/home/proj/coqdune/coq/_build/install/default/bin:/home/proj/coqdune/coq/_install_ci/bin:/home/jfehrle/.opam/4.07.1/bin:/home/jfehrle/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/mnt/c/windows/system32:/mnt/c/windows:/mnt/c/windows/System32/Wbem:/mnt/c/windows/System32/WindowsPowerShell/v1.0/:/mnt/c/windows/System32/OpenSSH/:/mnt/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NGX:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NvDLISR:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/Program Files/Docker/Docker/resources/bin:/mnt/c/ProgramData/DockerDesktop/version-bin:/mnt/c/Gradle/gradle-6.4.1/bin":/mnt/c/Users/jfehr/AppData/Local/Microsoft/WindowsApps:/snap/bin'
++ export COQBIN=/home/proj/coqdune/coq/_build/install/default/bin/
++ COQBIN=/home/proj/coqdune/coq/_build/install/default/bin/
++ ls -l /home/proj/coqdune/coq/_build/install/default/bin/
total 0
lrwxrwxrwx 1 jfehrle jfehrle 34 Apr 29 15:17 coq-tex -> ../../../default/tools/coq_tex.exe
lrwxrwxrwx 1 jfehrle jfehrle 39 Apr 29 15:18 coq_makefile -> ../../../default/tools/coq_makefile.exe
lrwxrwxrwx 1 jfehrle jfehrle 36 Apr 29 15:18 coqc -> ../../../default/topbin/coqc_bin.exe
lrwxrwxrwx 1 jfehrle jfehrle 35 Apr 29 15:18 coqc.byte -> ../../../default/topbin/coqc_bin.bc
lrwxrwxrwx 1 jfehrle jfehrle 35 Apr 29 15:18 coqchk -> ../../../default/checker/coqchk.exe
lrwxrwxrwx 1 jfehrle jfehrle 33 Apr 29 15:18 coqdep -> ../../../default/tools/coqdep.exe
lrwxrwxrwx 1 jfehrle jfehrle 38 Apr 29 15:18 coqdoc -> ../../../default/tools/coqdoc/main.exe
lrwxrwxrwx 1 jfehrle jfehrle 43 Apr 29 15:18 coqide -> ../../../default/ide/coqide/coqide_main.exe
lrwxrwxrwx 1 jfehrle jfehrle 37 Apr 29 15:18 coqidetop.byte -> ../../../default/ide/coqide/idetop.bc
lrwxrwxrwx 1 jfehrle jfehrle 38 Apr 29 15:18 coqidetop.opt -> ../../../default/ide/coqide/idetop.exe
lrwxrwxrwx 1 jfehrle jfehrle 37 Apr 29 15:18 coqpp -> ../../../default/coqpp/coqpp_main.exe
lrwxrwxrwx 1 jfehrle jfehrle 46 Apr 29 15:18 coqproofworker.opt -> ../../../default/topbin/coqproofworker_bin.exe
lrwxrwxrwx 1 jfehrle jfehrle 46 Apr 29 15:18 coqqueryworker.opt -> ../../../default/topbin/coqqueryworker_bin.exe
lrwxrwxrwx 1 jfehrle jfehrle 47 Apr 29 15:18 coqtacticworker.opt -> ../../../default/topbin/coqtacticworker_bin.exe
lrwxrwxrwx 1 jfehrle jfehrle 38 Apr 29 15:18 coqtop -> ../../../default/topbin/coqtop_bin.exe
lrwxrwxrwx 1 jfehrle jfehrle 42 Apr 29 15:18 coqtop.byte -> ../../../default/topbin/coqtop_byte_bin.bc
lrwxrwxrwx 1 jfehrle jfehrle 38 Apr 29 15:18 coqtop.opt -> ../../../default/topbin/coqtop_bin.exe
lrwxrwxrwx 1 jfehrle jfehrle 32 Apr 29 15:17 coqwc -> ../../../default/tools/coqwc.exe
lrwxrwxrwx 1 jfehrle jfehrle 37 Apr 29 15:18 coqworkmgr -> ../../../default/tools/coqworkmgr.exe
lrwxrwxrwx 1 jfehrle jfehrle 47 Apr 29 15:18 csdpcert -> ../../../default/plugins/micromega/csdpcert.exe
lrwxrwxrwx 1 jfehrle jfehrle 40 Apr 29 15:18 fake_ide -> ../../../default/ide/coqide/fake_ide.exe
lrwxrwxrwx 1 jfehrle jfehrle 38 Apr 29 15:18 ocamllibdep -> ../../../default/tools/ocamllibdep.exe
lrwxrwxrwx 1 jfehrle jfehrle 35 Apr 29 15:18 votour -> ../../../default/checker/votour.exe
++ CI_BUILD_DIR=/home/proj/coqdune/coq/_build_ci
++ CI_INSTALL_DIR=/home/proj/coqdune/coq/_install_ci
++ ls -l /home/proj/coqdune/coq/_build_ci
ls: cannot access '/home/proj/coqdune/coq/_build_ci': No such file or directory
++ true
++ declare -A overlays
++ set +x
+ git_download coqtail
+ local project=coqtail
+ local dest=/home/proj/coqdune/coq/_build_ci/coqtail
+ local giturl_var=coqtail_CI_GITURL
+ local giturl=https://github.com/whonore/Coqtail
+ local ref_var=coqtail_CI_REF
+ local ref=master
+ local ov_url=
+ local ov_ref=
+ '[' -d /home/proj/coqdune/coq/_build_ci/coqtail ']'
+ [[ -n '' ]]
+ '[' '' = 1 ']'
+ '[' '' = '' ']'
+ git clone https://github.com/whonore/Coqtail /home/proj/coqdune/coq/_build_ci/coqtail
Cloning into '/home/proj/coqdune/coq/_build_ci/coqtail'...
remote: Enumerating objects: 2496, done.
remote: Counting objects: 100% (76/76), done.
remote: Compressing objects: 100% (58/58), done.
remote: Total 2496 (delta 27), reused 44 (delta 16), pack-reused 2420
Receiving objects: 100% (2496/2496), 763.89 KiB | 2.21 MiB/s, done.
Resolving deltas: 100% (1480/1480), done.
+ cd /home/proj/coqdune/coq/_build_ci/coqtail
+ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
+ git log -n 1
commit 0d12ed5e9f36213d030ae9ce05d93942acf965d8 (HEAD -> master, origin/master, origin/HEAD)
Author: Wolf Honore <wolfhonore@gmail.com>
Date:   Thu Apr 29 08:16:52 2021 -0400

    Detect when coqtop crashes on start instead of hanging (#205)
+ [[ -n '' ]]
+ cd /home/proj/coqdune/coq/_build_ci/coqtail
+ PYTHONPATH=python
+ python3 -m pytest tests/coq
================================================== test session starts ===================================================
platform linux -- Python 3.6.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /home/proj/coqdune/coq/_build_ci/coqtail
collected 13 items

tests/coq/test_coqtop.py EEEEEEEEEEEEE                                                                             [100%]

========================================================= ERRORS =========================================================
___________________________________________ ERROR at setup of test_init_state ____________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
__________________________________________ ERROR at setup of test_rewind_start ___________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
_________________________________________ ERROR at setup of test_dispatch_rewind _________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
_____________________________________ ERROR at setup of test_dispatch_not_in_script ______________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
_______________________________________ ERROR at setup of test_query_same_state_id _______________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
____________________________________ ERROR at setup of test_option_different_state_id ____________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
________________________________________ ERROR at setup of test_dispatch_correct _________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
________________________________________ ERROR at setup of test_dispatch_unicode _________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
_________________________________________ ERROR at setup of test_goals_no_change _________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
__________________________________________ ERROR at setup of test_advance_fail ___________________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
________________________________ ERROR at setup of test_dispatch_ignore_comments_newlines ________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
______________________________________ ERROR at setup of test_recognize_not_option _______________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
_______________________________________ ERROR at setup of test_recognize_not_query _______________________________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        if ct.start(VERSION, None, None, "", [])[0] is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail("Failed to create Coqtop instance")
E           Failed: Failed to create Coqtop instance

tests/coq/test_coqtop.py:38: Failed
================================================ short test summary info =================================================
ERROR tests/coq/test_coqtop.py::test_init_state - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_rewind_start - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_dispatch_rewind - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_dispatch_not_in_script - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_query_same_state_id - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_option_different_state_id - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_dispatch_correct - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_dispatch_unicode - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_goals_no_change - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_advance_fail - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_dispatch_ignore_comments_newlines - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_recognize_not_option - Failed: Failed to create Coqtop instance
ERROR tests/coq/test_coqtop.py::test_recognize_not_query - Failed: Failed to create Coqtop instance
=================================================== 13 errors in 1.35s ===================================================
$ 
```

It seems to be a couple issues. The first is that Coqtail doesn't take `$COQBIN` into account by default and will just choose whatever it finds first on `$PATH` unless you tell it otherwise.

If we force it to look at `$COQBIN` and print some extra debug information
```patch
diff --git a/python/coqtop.py b/python/coqtop.py
index d891445..0bbbb15 100644
--- a/python/coqtop.py
+++ b/python/coqtop.py
@@ -81,10 +81,13 @@ class Coqtop:
         """Launch the Coqtop process."""
         assert self.coqtop is None

+        import os
+        coq_path = os.environ['COQBIN']
         self.xml = XMLInterface(version)
         self.logger.debug("start")
         try:
             launch = self.xml.launch(coq_path, coq_prog, filename, args)
+            print(f'COQBIN={os.environ["COQBIN"]} COQLIB={os.environ["COQLIB"]} {" ".join(launch)}')
             self.logger.debug(launch)
             self.coqtop = subprocess.Popen(  # pylint: disable=consider-using-with
                 launch,
diff --git a/tests/coq/test_coqtop.py b/tests/coq/test_coqtop.py
index 050b72e..5e3e4ca 100644
--- a/tests/coq/test_coqtop.py
+++ b/tests/coq/test_coqtop.py
@@ -31,11 +31,12 @@ def get_state(coq):
 def coq():
     """Return a Coqtop for each version."""
     ct = Coqtop()
-    if ct.start(VERSION, None, None, "", [])[0] is None:
+    out, _ = ct.start(VERSION, None, None, "", [])
+    if out is None:
         yield ct
         ct.stop()
     else:
-        pytest.fail("Failed to create Coqtop instance")
+        pytest.fail(f"Failed to create Coqtop instance: {out}")


 # Test Cases #
```

Then it finds the correct version, but fails with a different error
```
$ make ci-coqtail
... # a bunch of setup
+ cd /home/wolf/tmp/coqbug/coq/_build_ci/coqtail
+ PYTHONPATH=python
+ python3 -m pytest tests/coq
============================= test session starts ==============================
platform linux -- Python 3.8.8, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/wolf/tmp/coqbug/coq/_build_ci/coqtail
collected 13 items

tests/coq/test_coqtop.py EEEEEEEEEEEEE                                   [100%]

==================================== ERRORS ====================================
______________________ ERROR at setup of test_init_state _______________________

    @pytest.fixture(scope="function")
    def coq():
        """Return a Coqtop for each version."""
        ct = Coqtop()
        out, _ = ct.start(VERSION, None, None, "", [])
        if out is None:
            yield ct
            ct.stop()
        else:
>           pytest.fail(f"Failed to create Coqtop instance: {out}")
E           Failed: Failed to create Coqtop instance: Handshake with proofworker:0 failed: End_of_file

tests/coq/test_coqtop.py:39: Failed
---------------------------- Captured stdout setup -----------------------------
COQBIN=/home/wolf/tmp/coqbug/coq/_build/install/default/bin/ COQLIB=/home/wolf/tmp/coqbug/coq/_build/install/default/lib/coq /home/wolf/tmp/coqbug/coq/_build/default/ide/coqide/idetop.exe -main-channel stdfds -async-proofs on -async-proofs-command-error-resilience off -async-proofs-tactic-error-resilience off
... # more of the same
```

--- a/Makefile.ide
+++ b/Makefile.ide
@@ -19,6 +19,15 @@ IDEFILES=$(wildcard ide/*.lang) ide/coq_style.xml ide/coq.png) $(IDEBINDINGS)
 
 .PHONY: coqide
 
+ALL_IDE_FILES=$(shell find ide/coqide -name '*.ml' -or -name '*.mli' -or -name '*.mll' -or -name '*.c' -or -name '*.h')
+PROTO_IDE_FILES=$(shell find ide/coqide/protocol -name '*.ml' -or -name '*.mli' -or -name '*.mll' -or -name '*.c' -or -name '*.h')
+
+# We depend on all ide source files to trigger the dune rebuild
+$(COQIDE): $(ALL_IDE_FILES)
+
+# For idetop we depend on idetop.ml + all source files as it links to Coq
+$(IDETOP): $(ALL_ML_SOURCE_FILES) $(PROTO_IDE_FILES) ide/coqide/idetop.ml
+
 ifeq ($(HASCOQIDE),no)
 coqide:
 else

